### PR TITLE
Adds text components (HeadingText, TitleText, BodyText, ...)

### PR DESCRIPTION
**Pull request by Diogomartf**
## Summary

This PR adds text components to match our design system.

- add text components
- add them to UI page
- replaced all text with new Text components

Text components can receive this props: 
- `size` (1,2,3,etc)
- `classname` (pass styles, usually color: `"text-em-low"`)
- `weight` (accepts "medium", "semibold", "bold").
- `as` (can change the underling component by passing `as="h1"` or `as="div"` )

_(`OverlineText` does not receive `size` because has only 1 option)_

### preview of components in UI
image

### thought process

image

### Components in the wild

image
image

### CODE REVIEW

File: packages/app/ui/text/BodyText.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    size: {
+      3: ["text-md"],
+      2: ["text-sm"],
+      1: ["text-xs"],
+    },
+    weight: {
+      bold: ["font-bold"],
+      semibold: ["font-semibold"],
+      medium: ["font-medium"],
+    },
```
Comment (berteotti) [inline packages/app/ui/text/BodyText.tsx:15]: I see we are repeating this several times, should we have a style for weights only? We could then merge both styles with `cx`

File: packages/app/ui/text/TitleText.tsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+interface TitleTextProps extends TextProps {
+  size?: 1 | 2;
+}
+
+export const TitleText = ({
+  children,
+  size,
+  className,
+  weight,
+}: TitleTextProps) => (
```
Comment (berteotti) [inline packages/app/ui/text/TitleText.tsx:31]: Should we have `as` prop here too?

**Review by berteotti — COMMENTED** [review]
Looks good! Just made some small improvements comments :)

File: packages/app/ui/text/TitleText.tsx (hunk 2 above)
Comment (Diogomartf) [reply packages/app/ui/text/TitleText.tsx:31]: Yes. I'll fix this.

File: packages/app/ui/text/BodyText.tsx (hunk 1 above)
Comment (Diogomartf) [reply packages/app/ui/text/BodyText.tsx:15]: I'll explore this idea and get back to you.

Comment (Diogomartf) [reply packages/app/ui/text/BodyText.tsx]: refactored here: [27a9ce0](https://github.com/SwaprHQ/stackly-ui/pull/16/commits/27a9ce0dc09ff83ba0f3280f78e2eacf9338c1a7)

**Review by berteotti — DISMISSED** [review]
Look good to me. Don't forget to rebase it first.

**Review by berteotti — APPROVED** [review]

**Pull request merged**